### ISSUE TITLE
stack smashing detected ***: terminated

### ISSUE THREAD

**Issue by xor1ng** [golded-plus/golded-plus#29]
Hi,
I think this is due to the length of words without spaces. The message has a URL that is 397 characters long.

ex:
`https://yandex.ru/video/preview/?text=%D0%9F%D0%B8%D1%82%D0%B0%D0%BD%D0%B8%D0%B5%20%D1%81%D0%BC%D0%B0%D1%80%D1%82%D1%84%D0%BE%D0%BD%D0%B0%20%D0%BF%D1%80%D0%B8%20%D0%BF%D0%BE%D0%B4%D0%BA%D0%BB%D1%8E%D1%87%D0%B5%D0%BD%D0%BD%D0%BE%D0%BC%20%D0%9E%D0%A2%D0%93&path=wizard&parent-reqid=1635943451536303-567635306214991228-vla1-3230-vla-l7-balancer-8080-BAL-4128&wiz_type=vital&filmId=4353261120236309636`

PS:
GoldED+/LNX 1.1.5

**Comment by spanevin** [comment]
Steps to reproduce? Date of source code?

I just created a new message, inserted this URL and saved it successfully, no errors.

**Comment by spanevin** [comment]
I can see a message with this link, I can create a new message with this link in this or any other area, no errors.

GoldED+/LNX 1.1.5-b20220504 (Linux 5.15.52-gentoo iF6M10)
(but my system is x86, not x64)

**Comment by xor1ng** [comment]
I reproduced this bug on "5.4.0-122-generic #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022 x86_64".
Ok, I'll try it on x86

**Comment by xor1ng** [comment]
So, I can't build it on clean x86 ubuntu 18.04:
```
building gutlvers.o
gutlvers.cpp: Assembler messages:
gutlvers.cpp:572: Error: junk `(%edi)+4' after expression
gutlvers.cpp:573: Error: junk `(%edi)+8' after expression
gutlvers.cpp:601: Error: junk `(%edi)+4' after expression
gutlvers.cpp:602: Error: junk `(%edi)+8' after expression
gutlvers.cpp:608: Error: junk `(%edi)+4' after expression
gutlvers.cpp:609: Error: junk `(%edi)+8' after expression
../../GNUmakef.inc:31: recipe for target '../../obj/lnx/gall/gutlvers.o' failed
make[1]: *** [../../obj/lnx/gall/gutlvers.o] Error 1
make[1]: Leaving directory '/home/fido/golded-plus/goldlib/gall'
Makefile:49: recipe for target 'all' failed
make: *** [all] Error 2

```
```
# uname -a
Linux ubuntu-18 4.15.0-193-generic #204-Ubuntu SMP Fri Aug 26 19:17:55 UTC 2022 i686 i686 i686 GNU/Linux

```

```
# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/i686-linux-gnu/7/lto-wrapper
Target: i686-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.5.0-3ubuntu1~18.04' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=i686-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-targets=all --enable-multiarch --disable-werror --with-arch-32=i686 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=i686-linux-gnu --host=i686-linux-gnu --target=i686-linux-gnu
Thread model: posix
gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) 

```

**Comment by spanevin** [comment]
> gcc version 7.5.0

mine was built using gcc 11
my version has other issues, segfault with enabled spellchecker, no idea why :) nobody else could reproduce it... (unrelated to gcc version, it did segfault long before gcc 11 and latest source code)

**Comment by vitaliy-aksyonov** [comment]
Reproduced this with your message sample and memory sanitizer enabled. Will investigate and try to fix.

**Comment by vitaliy-aksyonov** [comment]
Fixed in  #62

**Comment by spanevin** [comment]
> So, I can't build it on clean x86 ubuntu 18.04:
> 
> ```
> building gutlvers.o
> gutlvers.cpp: Assembler messages:
> gutlvers.cpp:572: Error: junk `(%edi)+4' after expression
> gutlvers.cpp:573: Error: junk `(%edi)+8' after expression
> gutlvers.cpp:601: Error: junk `(%edi)+4' after expression
> gutlvers.cpp:602: Error: junk `(%edi)+8' after expression
> gutlvers.cpp:608: Error: junk `(%edi)+4' after expression
> gutlvers.cpp:609: Error: junk `(%edi)+8' after expression
> ../../GNUmakef.inc:31: recipe for target '../../obj/lnx/gall/gutlvers.o' failed
> make[1]: *** [../../obj/lnx/gall/gutlvers.o] Error 1
> make[1]: Leaving directory '/home/fido/golded-plus/goldlib/gall'
> Makefile:49: recipe for target 'all' failed
> make: *** [all] Error 2
> ```
> 
> ```
> # uname -a
> Linux ubuntu-18 4.15.0-193-generic #204-Ubuntu SMP Fri Aug 26 19:17:55 UTC 2022 i686 i686 i686 GNU/Linux
> ```
> 
> ```
> # gcc -v
> Using built-in specs.
> COLLECT_GCC=gcc
> COLLECT_LTO_WRAPPER=/usr/lib/gcc/i686-linux-gnu/7/lto-wrapper
> Target: i686-linux-gnu
> Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.5.0-3ubuntu1~18.04' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=i686-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-targets=all --enable-multiarch --disable-werror --with-arch-32=i686 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=i686-linux-gnu --host=i686-linux-gnu --target=i686-linux-gnu
> Thread model: posix
> gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04) 
> ```

This issue has been fixed as well, in #57
